components/infobox/wikis/apexlegends/infobox_map_custom.lua

```diff
@@ -0,0 +1,93 @@
+---
+-- @Liquipedia
+-- wiki=apexlegends
+-- page=Module:Infobox/Map/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Map = require('Module:Infobox/Map')
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local String = require('Module:StringUtils')
+local Title = require('Module:Infobox/Widget/Title')
+local Center = require('Module:Infobox/Widget/Center')
+
+local CustomMap = Class.new()
+local CustomInjector = Class.new(Injector)
+
+local _args
+local _map
+
+function CustomMap.run(frame)
+	local map = Map(frame)
+	_map = map
+	_args = map.args
+	map.addToLpdb = CustomMap.addToLpdb
+	map.createWidgetInjector = CustomMap.createWidgetInjector
+	return map:createInfobox(frame)
+end
+
+function CustomMap:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:addCustomCells(widgets)
+	table.insert(widgets, Cell{
+		name = 'Game Mode(s)',
+		content = {_args.gamemode}
+	})
+	local spanText = ''
+	if String.isNotEmpty(_args.spanstart) then
+		spanText = spanText .. ' - '
+	end
+	table.insert(widgets, Cell{
+		name = 'Played in ALGS',
+		content = {(_args.spanstart or '<i><b>Not </b></i>') .. spanText .. (_args.spanend or '<i><b>Currently</b></i>')}
+	})
+	if String.isNotEmpty(_args.ring) then
+		local ringTable = mw.html.create('table')
+			:addClass('wikitable wikitable-striped wikitable-bordered')
+			:css('width', '100%')
+			:css('text-align', 'center')
```

Comment:
Have you tried to use Module:Widget/Table?

Reply:
I've manage to create the table using the Module:Widget/Table. But I'm unsure as to how to table.insert it into "widgets" properly

https://liquipedia.net/apexlegends/Module:Infobox/Map/Custom/dev

I'm not sure cell will work given that the structure of the table is not created to then be converted "tostring()" 
I did try using the WidgetTable:make() but I'm unsure if that's correct.
Might be a little out my depth here, not 100% on how the widget classes work

Reply:
if it is a widget it should not need the cell widget around and hence no tostring

Reply:
i fixed most of it: https://liquipedia.net/apexlegends/index.php?title=Module%3AInfobox%2FMap%2FCustom%2Fdev&type=revision&diff=381582&oldid=381465

only the table display isn't working atm (at least on mobile)
so either
- i am to tired and missed something obvious or
- infobox and the table widget do not like each other
- we need some additional css/classes to be used

maybe @Rathoz can take a look when he is back next week

Reply:
![image](https://user-images.githubusercontent.com/3426850/185817368-33eb6c5e-53fc-417e-89b0-3689bd4abfd5.png)

Reply:
Update Widget/Table to set display:grid also.

Reply:
![image](https://user-images.githubusercontent.com/3426850/185817406-8eef6af4-32db-43f5-8ec2-0658e678cdb3.png)

Reply:
yeah that isn't easy to see on mobile^^
thx
adjusted the /dev accordingly
now only the row bg is an issue^^

Reply:
What's the issue with bg?

Reply:
At the moment the header bg colour is the same as the standard row bg colour.

Reply:
Then set a bg color :>

Reply:
1) would be hardcoded
2) originally the table was striped, i.e. alternating bg between rows, looks like that isn't supported with the widget table

Reply:
The Widget is using div's instead of an actual html table. The CSS only applies stripes to html tables

Reply:
yeah ik, but for div tables there exists `table-striped` as a class, see e.g.
https://liquipedia.net/commons/Module:DivTable

but that class doesn't have the bg colors as important
so the bg colors of the infobox stuff win

Reply:
So can this only be rectified by supporting the inclusion of the div classes? And marking backgrounds as important so they are not overwritten by the current infobox cell bg colour?

Reply:
I have had a look at the /dev module and modified it so as to get a good representation of how I think the table should look.

I've done this by using the `fo-nttax-infobox` class, and then manually setting the padding to remove it for this instance. 

I'm aware this probably isn't the best way of doing this, however it's been useful so as to visualise how the table is intended to look,
On top of which, it has the added benefit of restarting the striped counter, so as the header is always of a darker colour

https://liquipedia.net/apexlegends/Dark_meluca/KingsCanyonTest